tools/src/main/java/it/eng/tools/rest/api/ApplicationPropertiesAPIController.java

```diff
@@ -43,30 +47,30 @@ public ApplicationPropertiesAPIController(ApplicationPropertiesService service,
 	 */
 
 	@GetMapping(path = "/")
-	public ResponseEntity<List<JsonNode>> getProperties(@RequestParam(required = false) String key_prefix) {
+	public ResponseEntity<GenericApiResponse<List<JsonNode>>> getProperties(@RequestParam(required = false) String key_prefix) {
 		log.info("getProperties()");
 		if(key_prefix != null && !key_prefix.isBlank()) log.info(" with key_prefix " + key_prefix);
 		var properties = propertiesService.getProperties(key_prefix);
 
 		applicationEventPublisher.publishEvent(properties);
 
 		return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON)
-				.body(Serializer.serializeProtocolListOfJsonNode(properties));
+				.body(GenericApiResponse.success(Serializer.serializeProtocolListOfJsonNode(properties), "ok", HttpStatus.OK.value()));
 	}
 
 	@GetMapping(path = "/{key}")
-	public ResponseEntity<JsonNode> getPropertyByKey(@PathVariable String key) {
+	public ResponseEntity<GenericApiResponse<JsonNode>> getPropertyByKey(@PathVariable String key) {
 		log.info("Fetching property with key " + key);
 
 		ApplicationProperty property = propertiesService.getPropertyByKey(key).orElseThrow(() -> new ApplicationPropertyNotFoundAPIException("Property with key " + key + " not found"));

```

Comment:
Please move this throw exception into service layer, in controller just call service.getProeprtyByKey() and in service do this check if property is found or not